framework/doc/sqa/SoftwareRequirements.tex

```diff
@@ -0,0 +1,454 @@
+result in a user passing on the platform. Ultimately, the decision to utilize a framework comes down to whether or
+not you trust the code in the framework and those developing it to be able to support your desired use-case.
+No matter the technical capabilities of a framework, without trust users will look elsewhere. This is especially
+true of those not trained in software development or computational science.
+
+Developing trust in a framework goes beyond utilizing ``best practices'' for the code developed, it is equally
+important that the framework itself is built upon tools that are trusted. For this reason, MOOSE relies on a
+well-established code base of libMesh and PETSc.
+The libMesh library provides foundational capability for the finite element method and provides interfaces to
+leading-edge numerical solution packages such as PETSc.
+
+With these principles in mind, an open source, massively parallel, finite element, multiphysics framework has been conceived.
+MOOSE is an on-going project started in 2008 aimed toward a common platform for creation of new multiphysics tools.
+This document provides design details pertinent to application developers as well as framework developers.
+
+\subsection{Use Cases}
+The MOOSE Framework is targeted at two main groups of actors: Developers and Users. Developers are the main
+use case. These are typically students and professionals trained in science and engineering fields with some
+level of experience with coding but typically very little formal software development training. The other user
+group is Users. Those who intend to use an application built upon the framework without writing any computer
+code themselves. Instead they may modify or create input files for driving a simulation, run the application, and
+analyze the results. All interactions through MOOSE are primarily through the command-line interface and through
+a customizable block-based input file described later (REFERENCE NEEDED).
+
+\subsection{Software Overview}
+
+\subsubsection{Introduction}
+The MOOSE framework itself is composed of a wide range of plugable systems. Each system is generally composed of a single or small set of C++ objects intended to be specialized by a Developer to solve a specific problem. To accomplish this design goal, MOOSE uses several modern object-oriented design patterns. The primary overarching pattern is the ``Factory Pattern''. User's needing to extend MOOSE may inherit from one of the public interfaces overridden the designated methods to accomplish a specific task. The design of each of these systems is documented on the mooseframework.org wiki in the Tutorial section. Additionally, up-to-date documentation extracted from the source is maintained on the the mooseframework.org documentation site after every successful merge to MOOSE's stable branch. After these objects are created, the can be registered with the framework and used immediately in a MOOSE input file.
```

Comment:
This sentence makes no sense.
"may inherit from one of the public interfaces overridden the designated methods to accomplish a specific task"